specutils/io/parsing_utils.py

```diff
@@ -99,18 +99,17 @@ def spectrum_from_column_mapping(table, column_mapping, wcs=None):
             kwarg_val = u.Quantity(table[col_name], tab_unit)
 
             # Attempt to convert the table unit to the user-defined unit.
-            log.debug("Attempting auto-convert of table unit '%s' to "
-                          "user-provided unit '%s'.", tab_unit, cm_unit)
+            log.debug(f"Attempting auto-convert of table unit {tab_unit} to "
+                      f"user-provided unit {cm_unit}.")
```

Comment:
The logging library in python is optimized to use `%s` formatting ([ref](https://docs.python.org/3/howto/logging.html#optimization), [ref](https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles)), which is why you'll find it throughout the codebase. I realize it's not strictly enforced, so this is probably a fine change (easier to parse > optimization), but just for future reference.

Reply:
Interesting... but does this line run so much that it warrants optimization?

```python
In [14]: %timeit log.debug(f"'{x.unit}'")
3.59 µs ± 176 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [15]: %timeit log.debug("'%s'" % x.unit)
1.49 µs ± 5.85 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
It's actually `log.debug("'%s'", x.unit)` which is probably using `.format()` internally:
```python
In [8]: %timeit log.debug("'%s'", x.unit)
526 ns ± 8.8 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [9]: %timeit log.debug("'%s'".format(x.unit))
611 ns ± 11.2 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```
Still, the speedup in this context is very modest compared to the time the actual code takes
```python
In [10]: %timeit x.to(u.nm)
10.6 µs ± 199 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [11]: %timeit x.to(u.Hz, equivalencies=u.spectral())
64.5 µs ± 2.22 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```
I find the original version easy enough to parse; just needs to be known since f-strings are otherwise strongly endorsed now in astropy core.

Reply:
@dhomeier is exactly right, the logging package is built to be passed arguments which are evaluated as-needed. In the case of f-strings, interpolation happens before logging visibility criteria is evaluated, which can lead to cases where string errors are raised, nothing gets logged, and the program exits because of the automatic evaluation of `__str__`. In `%s`-style, interpolation happens within the logging package and only as needed. It will therefore be caught as logging-related issue, raises an error, and the code will continue executing.

Additionally, log aggregation (i.e. the sentry package), works because the `%s` string template is the same between similar logging statements. Obviously, we don't make prolific use of of this, nor do we experience string-related log-avoiding bugs, but there is a reason that python did not switch the package to using `{}` or f-style string interpolation with python 3.

Reply:
FWIW, just because `astropy` uses it doesn't mean `specutils` has to. I don't think `astropy` uses much `logging` internally.

Though you probably want to document this somewhere in your dev docs, if not already.

Reply:
No, I also took this solely as a recommendation, but still think it a good idea to synchronise the styles of coordinated packages unless there are more important considerations against it like here.

Reply:
> Though you probably want to document this somewhere in your dev docs, if not already.

I don't think the dev docs here include anything on code style; might be best to just link to https://docs.astropy.org/en/latest/development/codeguide.html
But I believe the performance considerations could be of value there, too, even if logging is currently used very little (found some direct usage of `logging` in the FITS scripts but hardly any of its custom `AstropyLogger`) – maybe just adding a note to the recommendation on f-strings.

Reply:
I wouldn't bring `AstropyLogger` into this... 😆 There was a separate discussion, which resulted in doc update clarifying that the logger is for `astropy` internal use only.

If you want to put the doc upstream in `astropy` dev docs, I'll have to loop in @embray here since he's working on dev docs revamp over there. They would have a better idea whether this particular doc should go.

Reply:
I had no intention to switch to `AstropyLogger` here, was just looking where `logging` is used inside astropy, if at all.
If it isn't, there's obviously not much need to mention its own formatting interface. But it probably could not hurt either...